src/prototype.rs

```diff
@@ -45,12 +45,18 @@ enum ItemBody {
     Inline(usize, bool, bool), // Perhaps this should be MaybeEmphasis?
     MaybeCode(usize),
     MaybeHtml,
+    MaybeLinkOpen,
```

Comment:
Would it be nice to at some point have different enums for the pre- and post resolution stage?

Reply:
I'm not sure that's really possible, when you're in the middle of processing you're going to have some of both. The order nodes are processed is somewhat chaotic because of the requirements of the spec, it's not like a simple tree traversal where you can take a tree of one type as input and produce a tree of another type.